### ISSUE TITLE
How Do I upload?

### ISSUE THREAD

**Issue by Heineken** [niccokunzmann/mundraub-android#301]
Spent the afternoon adding trees to the app. First with photos and description, later without because I didn't know what to describe.
Later I logged in and looked for the update/sync button. To no avail. I read about the upload function in various places, but no how-to. In My Trees in the single view there was no upload option either (because the description was missing, as I understood later).

So, two things that annoy me here:
1. I have to add a (dummy) description
2. I have to upload all the entries one by one

PROPOSED FIX (one of these):
- add a manual explaining how to upload
- add an upload option/button in the top level menu. Pressing it should upload all the new entries.
- add a message explaining when single upload is not possible why not (#300)

**Comment by Heineken** [comment]
>     * Will all plants be uploaded?
Yes, If the user doesn't want that, he shouldn't press the button and upload single plants instead.
Changes are tricky (gotta check for conflicts), but they are not implemented anyway. Currently, if am not mistaken, changes made offline cannot be uploaded, right? (#7)

> What if I have 100 plants and only see the first 10?

You mean in the list? Doesn't matter. There could be confirmation dialogue before the actual upload: "Upload 24 plants? OK/Cancel"

> Will the last ones also be uploaded?

Sure. All the new ones that haven't been uploaded yet, which is tracked apparently as I see an online indicator.

> What if I forgot to upload a plant?

I don't get the question.

> There are several cases after pressing the button for upload:     
>  plant is uploaded
>  upload failed due to some api error
>  the location of the plant should be confirmed because (1) it is a GPS location or (2) there is a similar plant close to it
> the plant is missing some information

-> #300 

> Seeing this, I would like to see
> 
> an "upload all plants" button and when you press it, you see a list which is getting longer, 

I'd rather keep seeing the list of my plants and online indicators appearing.

> and a progress bar and text "uploading X plants"

👍

> new plants are added to the list with an description of why the upload failed

No, old list with new error description, if needed.

> The log-in view opens when the user is not logged in, yet.

👍
 
> What do you think?

Do you know StreetComplete (on [GitHub](https://github.com/westnordost/StreetComplete), [FDroid](https://f-droid.org/en/packages/de.westnordost.streetcomplete/))
It's a well-made interface to easily improve, complete or add OSM data. It bears some similarities. Could be a source of ideas or even code.
Or, thinking further, why not put the data on OSM and use all the available tools? I know there's a tree tag...
Well, you asked what I think ... :-)

**Comment by niccokunzmann** [comment]
> Yes, If the user doesn't want that, he shouldn't press the button and upload single plants instead.

I agree

> changes made offline cannot be uploaded, right? (#7)

They can be uploaded to na-ovoce, not to mundraub. One would need to reverse-engineer that. Should not be too hard.
https://www.youtube.com/watch?v=kNM7MtFgk6k

> No, old list with new error description, if needed.

So, you would modify the existing list and add a text field to each plant with an error, so that you can read something like 

    picture       1x apple
    Upload failed because there is an apple nearby. Press the "upload all plants" button again to upload the apple either way.
    ---------------------------
    picture      5x walnut   [www-symbol]
    upload successful
    ---------------------------
    picture      hazelnut
    [rotating icon] upload in progress

> Do you know StreetComplete (on GitHub, FDroid)

No, I do not. It seems quite nice but does not run on any of my devices.

**Comment by Heineken** [comment]
> > changes made offline cannot be uploaded, right? (#7)
> 
> They can be uploaded to na-ovoce, not to mundraub. One would need to reverse-engineer that. Should not be too hard.

OK. Sounds tempting.

> https://www.youtube.com/watch?v=kNM7MtFgk6k
>
Nice talk. Nice Firefox tool. I have used occasionally, but when I really needed such a thing, it didn't exist yet.
>
> So, you would modify the existing list and add a text field to each plant with an error, so that you can read something like
> 
> ```
> picture       1x apple
> Upload failed because there is an apple nearby. Press the "upload all plants" button again to upload the apple either way.
> ---------------------------
> picture      5x walnut   [www-symbol]
> upload successful
> ---------------------------
> picture      hazelnut
> [rotating icon] upload in progress
> ```
Exactly!

> 
> > Do you know StreetComplete (on GitHub, FDroid)
> No, I do not. It seems quite nice but does not run on any of my devices.

Oh, how odd. I have used it on 3 different phones.
What are your devices? Smart-TV? C64? Rooted phone running minimal Linux?
Maybe you were too restrictive with the permissions with XPrivacy or your firewall or something.
